### PR TITLE
LPS-190642 | Test Automation | Master

### PR DESCRIPTION
**Pull request by Luizcarlosqueiroz**
**Ticket:** [LPS-190642](https://liferay.atlassian.net/browse/LPS-190642)

### CODE REVIEW

**Comment by Luizcarlosqueiroz** [comment]
ci:test:sf

**Comment by Luizcarlosqueiroz** [comment]
Hey @patriciajeanperez @timpak !
I had to make some changes from https://github.com/liferay-bpm-qa/liferay-portal/pull/958 to allow `'` . I tested with @thaiscabrals , and she has able to use the macro successfully in the test she's writing.
Sorry for that.

File: modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro [hunk 1]
```diff
@@ -204,6 +204,20 @@ definition {
...
 
+		if (!(isSet(isReadOnly))) {
+			var isReadOnly = "false";
+		}
+
+		if (isSet(readOnlyExpression)) {
+			var single_quote = "'";
+			var scape_quotes = "'\''";
+
+			var readOnlyExpression = StringUtil.replace(${readOnlyExpression}, ${single_quote}, ${scape_quotes});
```
Comment (Luizcarlosqueiroz) [inline modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:215]: I tried to implement without using the `single_quote` and `scape_quotes`, like this:

- var readOnlyExpression = StringUtil.replace(${readOnlyExpression}, "`'`", "`'\''`");

But according to the errors shown at terminal, it's treating as many strings arguments, not only the three necessary. The solution I came up was using variables.

**Comment by thaiscabrals** [comment]
Just starting reviewing :)

**Comment by patriciajeanperez** [comment]
ci:forward

**Pull request closed without merging**